starter-code/database.js

```diff
@@ -24,105 +23,116 @@ class Database {
     }
   }
 
-  close(callback = (error) => {}){
-    if (this.database){
+  close(callback = error => {}) {
+    if (this.database) {
       this.database.close(true, callback);
     } else {
       callback();
     }
   }
   // Insert a user
   // user is the object to insert into the collection
-  // callback has two arguments error and result 
-  insertUser(user, callback = (error, result) => {}){
+  // callback has two arguments error and result
+  insertUser(user, callback = (error, result) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         // LAB 1
         // Implement the query to insert a user
         // user is the document that we want to insert
         // remeber once it's finish to comment callback('Error inserting user');
-        
-        callback('Error inserting user');
+        database.collection("users").insertOne(user, callback);
+
+        // callback("Error inserting user");
       }
     });
   }
 
-  listUsers(callback = (error, users) => {}) {
+  listUsersDisplay(callback = (error, users) => {}) {
     this.connect((error, database) => {
-      if (error){
+      if (error) {
         callback(error);
       } else {
         //  LAB 2
         // Implement the query to insert a user
         // remeber once it's finish to comment callback('Error listing users');
-        
-        callback('Error listing users');
+        callback(null, database.collection("users").find());
```

Comment:
If you do it this way, the value that you give to the callback is what's returned by the `find` method. This is not what you want because the `find` method is asynchronous. It doesn't return the result of the requests; it expects you to tell it what to do with the result by passing a callback function.

The solution is therefore: `database.collection("users").find({}, callback)`.